### ISSUE TITLE
CRI-O metrics file redirects to an html page hence it is failing to be deployed with kubectl

### ISSUE THREAD

**Issue by thanos1983** [cri-o/cri-o#5314]
**Description**

This ticket is related to [Error converting YAML to JSON: yaml: line 124: mapping values are not allowed in this context #5285](https://github.com/cri-o/cri-o/issues/5285#issuecomment-916820681).

After a bit of looking around I figured out that there is a redirect in the background that is happening and the last jump is an html page. Sample below:

```bash
✗ curl -L https://github.com/cri-o/cri-o/blob/master/contrib/metrics-exporter/cluster.yaml | tail -15
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100   144  100   144    0     0   1920      0 --:--:-- --:--:-- --:--:--  1920
100  155k    0  155k    0     0   356k      0 --:--:-- --:--:-- --:--:--  674k
    <path fill-rule="evenodd" d="M5.75 1a.75.75 0 00-.75.75v3c0 .414.336.75.75.75h4.5a.75.75 0 00.75-.75v-3a.75.75 0 00-.75-.75h-4.5zm.75 3V2.5h3V4h-3zm-2.874-.467a.75.75 0 00-.752-1.298A1.75 1.75 0 002 3.75v9.5c0 .966.784 1.75 1.75 1.75h8.5A1.75 1.75 0 0014 13.25v-9.5a1.75 1.75 0 00-.874-1.515.75.75 0 10-.752 1.298.25.25 0 01.126.217v9.5a.25.25 0 01-.25.25h-8.5a.25.25 0 01-.25-.25v-9.5a.25.25 0 01.126-.217z"></path>
</svg>
      <svg aria-hidden="true" height="16" viewBox="0 0 16 16" version="1.1" width="16" data-view-component="true" class="octicon octicon-check js-clipboard-check-icon color-text-success d-none m-2">
    <path fill-rule="evenodd" d="M13.78 4.22a.75.75 0 010 1.06l-7.25 7.25a.75.75 0 01-1.06 0L2.22 9.28a.75.75 0 011.06-1.06L6 10.94l6.72-6.72a.75.75 0 011.06 0z"></path>
</svg>
    </clipboard-copy>
  </div>
</template>




  </body>
</html>

```

**Steps to reproduce the issue:**
1. Simply apply the curl redirect `curl -L https://github.com/cri-o/cri-o/blob/master/contrib/metrics-exporter/cluster.yaml` and observe the output.
2. If the user downloads the url page and tries to deploy e.g. `kubectl create -f cluster.yaml` it will get the following error:

```bash
Error loading resource_definition: mapping values are not allowed here line 124, column 65
```

Which is very close to what we observe on the related ticket.
3.

**Describe the results you received:**
We are getting redirected to an html file instead of the actual file.

**Describe the results you expected:**
To be redirected to the actual `cluster.yaml` file.

**Additional information you deem important (e.g. issue happens only occasionally):**
Not related to CRI-O version.

**Output of `crio --version`:**
Not related to CRI-O version.
```
(paste your output here)
```

**Additional environment details (AWS, VirtualBox, physical, etc.):**
Bare metal deployment but not related also.

**Comment by haircommander** [comment]
I think the URL you're using isn't correct:
https://raw.githubusercontent.com/cri-o/cri-o/main/contrib/metrics-exporter/cluster.yaml
should not redirect

do we have the wrong URL in our docs somewhere?

**Comment by thanos1983** [comment]
> I think the URL you're using isn't correct:
> https://raw.githubusercontent.com/cri-o/cri-o/main/contrib/metrics-exporter/cluster.yaml
> should not redirect
> 
> do we have the wrong URL in our docs somewhere?

Hello @haircommander ,

I did not pick up the url from the documentation I just followed the path from the git hub.

Sample: `https://github.com/cri-o/cri-o/tree/main/contrib/metrics-exporter` and then `cri-o/contrib/metrics-exporter/cluster.yaml` it redirects to `https://github.com/cri-o/cri-o/blob/main/contrib/metrics-exporter/cluster.yaml` from there when I try to reach it through curl there is another redirect happening in the background.

The goal for me was to be able to pull and deploy the latest version of the file (in case it would be updated in future).

Sample: `kubectl create -f <url>`, without the need to download the file.

So maybe we should add the documentation here [Exporting Metrics via Prometheus](https://github.com/cri-o/cri-o/blob/main/tutorials/metrics.md#exporting-metrics-via-prometheus) regarding this url?

I think we can close the ticket, this is just a suggestion. Now that I have the correct url it works fine.
